Master

swardmedia opened the pull request.

swardmedia committed:
Update README.md

Initial info after fork.

swardmedia committed:
:space_invader: Added .gitattributes & .gitignore files

swardmedia committed:
Update README.md

swardmedia committed:
Initial upload of modernGrid

The initial upload of modernGrid, with examples

swardmedia committed:
Delete .gitattributes

swardmedia committed:
Delete .gitignore

swardmedia committed:
Update README.md

Adding in demo URL

swardmedia committed:
Update README.md

fix typo

swardmedia committed:
Update README.md

swardmedia committed:
Update README.md

pblanchardie commented:
Nice fork, thanks! I think it could be nice to merge, let's check with @reconstrukt (Matthew).
The styling is quite old, it should be nice to have a Bootstrap version or something more sexy...

reconstrukt commented:
Hey Patrice -

Take a look at the options he's got.  Feels like a lot of dependencies? Which is fine, but I'm thinking he should re-organize the code. His current PR is basically the Ingrid.js fork, plus 40 other dependent files :)

Thoughts?
Matt 

> On Jun 27, 2016, at 9:35 AM, Patrice Blanchardie notifications@github.com wrote:
> 
> Nice fork, thanks! I think it could be nice to merge, let's check with @reconstrukt (Matthew).
> The styling is quite old, it should be nice to have a Bootstrap version or something more sexy...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

swardmedia committed:
Organization update

Updated structure to differentiate between plugin source and included
examples.

reconstrukt commented:
Thanks @swardmedia, better organized

Also suggest the dependencies (such as editinplace, bootbox, etc) be optional as part of a grid instance config, and the code should only initialize if the requisite libs are available.  The original Ingrid takes a similar approach with the jquery cookie plugin (for storing grid state).

Also, let's exclude any of those plugin files from the source, and reference them via a grunt file

reconstrukt commented:
**Edit

**Let's exclude plugins from source and reference them as dependencies in a bower file

And, re-opening.

swardmedia committed:
Removed examples from github - examples can be downloaded with source from http://swardmedia.com/moderngrid
Updated basic css to bring it closer to modern frameworks (bootstrap)
Added edit in place functionality - no external plugin needed.

Point of clarification: ModernGrid does NOT require additional plugins - the only dependency is jQuery.  Examples have been added to demonstrate full functionality without plugins - http://swardmedia.com/moderngrid/fullExampleNoPlugins.html and http://swardmedia.com/moderngrid/rightClickNoPlugins.html.

Plugins are only used in the examples to show ModernGrid working alongside other plugins within a modern framework such as bootstrap.

Joe Sward

swardmedia committed:
Fixed error in edit in place - now setting attribute value, not attribute val
